### PR TITLE
Custom field values now display in payment details.

### LINKED ISSUE #5621: Custom field values should be displayed in the payment details
## Acceptance Criteria

- [ ] Custom field values are visible in the payment details

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #5621 

## Description

- Adds a new metabox to the payment details screen for displaying custom field values.

## Visuals

image

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php [hunk 1]
```diff
@@ -19,8 +19,8 @@ function( $donationID, $donationData ) use ( $hook ) {
 				do_action( "give_fields_$hook", $fieldCollection, $donationData['give_form_id'] );
 				$fieldCollection->walk(
 					function( $field ) use ( $donationID ) {
-						if ( isset( $_POST[ 'give_' . $field->getName() ] ) ) {
-							$value = wp_strip_all_tags( $_POST[ 'give_' . $field->getName() ], true );
+						if ( isset( $_POST[ $field->getName() ] ) ) {
+							$value = wp_strip_all_tags( $_POST[ $field->getName() ], true );
```
Comment (kjohnson) [inline src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php:23]: This fixes a bug in persisting the custom field value - the prefix was removed from the field name but not the logic.

File: src/Form/LegacyConsumer/ServiceProvider.php [hunk 2]
```diff
@@ -27,8 +27,9 @@ function() {
 	 */
 	public function boot() {
 		give( TemplateHooks::class )->walk( give( Commands\SetupNewTemplateHook::class ) );
-		give( TemplateHooks::class )->walk( give( Commands\SetupFieldPersistance::class ) );
 		give( TemplateHooks::class )->walk( give( Commands\SetupFieldValidation::class ) );
+		give( TemplateHooks::class )->walk( give( Commands\SetupFieldPersistance::class ) );
+		give( TemplateHooks::class )->walk( new Commands\CommandFactory( Commands\SetupPaymentDetailsDisplay::class ) );
```
Comment (kjohnson) [inline src/Form/LegacyConsumer/ServiceProvider.php:32]: I plan to expand use of the `CommandFactory` to other commands to cleanup the nested callbacks.

**Comment by kjohnson** [comment]
@JasonTheAdams is this one good to go?

**Comment by JasonTheAdams** [comment]
@kjohnson Were you going to make the change we discussed to its not creating N instances that bind to the action N times for every hook?

**Comment by kjohnson** [comment]
@JasonTheAdams ah, yeah, I was going to do that is a refactor PR since it touches multiple commands.

**Review by JasonTheAdams — APPROVED** [review]
Great work, @kjohnson! As you know, we'll want to do some optimization. But otherwise, I noticed that each field makes its own metabox. That will get visually cumbersome for forms with many custom fields. We may consider a "Custom Fields" metabox for now where they all live in. But given how the code is structured that will likely need to be after some optimization. So let's merge this and move along for now.

**Pull request merged**